jwst/source_catalog/source_catalog_step.py

```diff
@@ -44,6 +44,9 @@ class SourceCatalogStep(Step):
     reference_file_types = ['apcorr', 'abvegaoffset']
 
     def process(self, input_model):
+        if not isinstance(self.bkg_boxsize, int):
+            raise TypeError('bkg_boxsize must be an integer')
+
```

Comment:
`stpipe` should be doing all input parameter validation.  If it is not, we should fix it in `stpipe`.

Reply:
There is a general issue of parameter values not being validated after `Step.__init__`:

```
In [1]: from jwst.source_catalog import SourceCatalogStep

In [2]: step = SourceCatalogStep()

In [3]: step.bkg_boxsize
Out[3]: 100

In [4]: type(step.bkg_boxsize)
Out[4]: int

In [5]: step.bkg_boxsize = "100"

In [6]: type(step.bkg_boxsize)
Out[6]: str

In [7]: step = SourceCatalogStep(bkg_boxsize="100")

In [8]: type(step.bkg_boxsize)
Out[8]: int

In [9]: step = SourceCatalogStep(bkg_boxsize="200")

In [10]: type(step.bkg_boxsize)
Out[10]: int

In [11]: step.bkg_boxsize
Out[11]: 200

In [12]: step = SourceCatalogStep(bkg_boxsize="foo")
---------------------------------------------------------------------------
ValidationError                           Traceback (most recent call last)
<ipython-input-12-1dbadfbdbc35> in <module>
----> 1 step = SourceCatalogStep(bkg_boxsize="foo")

~/miniconda3/envs/jwst/lib/python3.9/site-packages/stpipe/step.py in __init__(self, name, parent, config_file, _validate_kwds, **kws)
    295         if _validate_kwds:
    296             spec = self.load_spec_file()
--> 297             kws = config_parser.config_from_dict(
    298                 kws, spec, root_dir=dirname(config_file or ''))
    299 

~/miniconda3/envs/jwst/lib/python3.9/site-packages/stpipe/config_parser.py in config_from_dict(d, spec, root_dir, allow_missing)
    272 
    273     if spec:
--> 274         validate(config, spec, root_dir=root_dir, allow_missing=allow_missing)
    275     else:
    276         config.walk(string_to_python_type)

~/miniconda3/envs/jwst/lib/python3.9/site-packages/stpipe/config_parser.py in validate(config, spec, section, validator, root_dir, allow_missing)
    361 
    362         if len(messages):
--> 363             raise ValidationError('\n'.join(messages))
    364     finally:
    365         config.main.configspec = orig_configspec

ValidationError: Config parameter 'bkg_boxsize': the value "foo" is of the wrong type.
```

And that issue is discussed in #2159 and is not unique.  The casting of string to int is perhaps a feature of ConfigObj?  Not sure.

Reply:
@jdavies-st I removed the explicit type check.